dashboard/test/factories/factories.rb

```diff
@@ -3,6 +3,16 @@
 FactoryGirl.allow_class_lookup = false
 
 FactoryGirl.define do
+  factory :course_version do
+    sequence(:key) {|n| "202#{n-1}"}
+    sequence(:display_name) {|n| "2#{n-1}-2#{n}"}
+    with_unit_group
+
+    trait :with_unit_group do
```

Comment:
Took this pattern from https://gist.github.com/traviskroberts/2830535#gistcomment-2791342

Reply:
would it make sense to define `with_unit` now too? otherwise it's confusing as to why we are defining a trait and then setting it unconditionally.

Reply:
Added association + updated factory and test.

Reply:
Looks great!